packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx

```diff
@@ -0,0 +1,125 @@
+import { generateQueryKey, RequestKey } from '../../../lib/query';
+import { disabledRefetch } from '../../../lib/func';
+import { TOP_READER_BADGE } from '../../../graphql/users';
+import { downloadUrl } from '../../../lib/blob';
+import LogContext from '../../../contexts/LogContext';
+import { LogEvent, TargetId, TargetType } from '../../../lib/log';
+
+const TopReaderBadgeModal = (
+  props: ModalProps & {
+    onAfterClose: (keywordValue: string) => void;
+    onAfterOpen: (keywordValue: string) => void;
+  },
+): ReactElement => {
+  const { onRequestClose, onAfterOpen, onAfterClose } = props;
+
+  const { user } = useAuthContext();
+  const { logEvent } = useContext(LogContext);
+  const isMobile = useViewSize(ViewSize.MobileL);
+
+  const { data: topReaderBadge } = useQuery({
+    queryKey: generateQueryKey(RequestKey.TopReaderBadge),
+    queryFn: async () => {
+      return (
+        await gqlClient.request<{ topReaderBadge: TopReader[] }>(
+          TOP_READER_BADGE,
+          {
+            limit: 1,
+          },
+        )
+      ).topReaderBadge[0];
+    },
+    ...disabledRefetch,
+  });
+
+  const { mutateAsync: onDownloadUrl, isPending: downloading } = useMutation({
+    mutationFn: downloadUrl,
+  });
+
+  const onClickDownload = useCallback(async () => {
+    if (!topReaderBadge) {
+      return;
+    }
+
+    const formattedDate = new Date(topReaderBadge.issuedAt).toLocaleString(
+      'en-US',
+      {
+        year: 'numeric',
+        month: 'long',
+      },
+    );
+
+    await onDownloadUrl({
+      url: topReaderBadge.image,
+      filename: `${formattedDate} Top Reader in ${topReaderBadge.keyword.flags.title}.png`,
+    });
+
+    logEvent({
+      event_name: LogEvent.TopReaderBadgeDownload,
+      target_type: TargetType.Badge,
+      target_id: TargetId.TopReader,
+      extra: JSON.stringify({
+        tag: topReaderBadge.keyword.value,
+      }),
+    });
+  }, [logEvent, onDownloadUrl, topReaderBadge]);
+
+  if (!topReaderBadge) {
+    return null;
+  }
+
+  return (
+    <Modal
+      {...props}
+      size={ModalSize.Small}
+      isDrawerOnMobile
+      onAfterClose={() => onAfterClose('keyword.value')}
+      onAfterOpen={() => onAfterOpen('keyword.value')}
+    >
+      <Modal.Body className="flex flex-col items-center justify-center gap-4 text-center">
+        <h1 className="font-bold typo-title1">
+          You&apos;ve earned the top reader badge!
+        </h1>
+        <TopReaderBadge
+          user={user}
+          keyword={topReaderBadge.keyword}
+          issuedAt={topReaderBadge.issuedAt}
+        />
+
+        <Button
+          className={classNames('w-full', !isMobile && 'max-w-80')}
+          variant={ButtonVariant.Primary}
+          icon={<DownloadIcon secondary />}
+          disabled={!topReaderBadge.image}
+          loading={downloading}
+          onClick={() => onClickDownload()}
+        >
+          Download badge
+        </Button>
+
+        {!isMobile && (
```

Comment:
Why do we need this specific condition, the modals have their generic use-case for mobile/desktop buttons already right?

Reply:
[In the design](https://www.figma.com/design/nmfWPS7x3kzLUvYMkBx2kW/daily.dev---Dev-Mode?node-id=14437-23460&node-type=instance&m=dev), the close button is at the bottom. Usually they're a X button at the top right. And when it's on mobile, it already comes with close button on the drawer, so I need to hide it.

Reply:
I can set `displayCloseButton: false` on `drawerProps` and then always render this button instead.

Reply:
```diff
diff --git a/packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx b/packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx
index e535994f..578d9a7d 100644
--- a/packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx
+++ b/packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx
@@ -86,6 +86,9 @@ const TopReaderBadgeModal = (
       isDrawerOnMobile
       onAfterClose={() => onAfterClose(topReaderBadge.keyword.value)}
       onAfterOpen={() => onAfterOpen(topReaderBadge.keyword.value)}
+      drawerProps={{
+        displayCloseButton: false,
+      }}
     >
       <Modal.Body className="flex flex-col items-center justify-center gap-4 text-center">
         <h1 className="font-bold typo-title1">
@@ -108,15 +111,13 @@ const TopReaderBadgeModal = (
           Download badge
         </Button>
 
-        {!isMobile && (
-          <Button
-            className="w-full max-w-80"
-            variant={ButtonVariant.Float}
-            onClick={onRequestClose}
-          >
-            Close
-          </Button>
-        )}
+        <Button
+          className={classNames('w-full', !isMobile && 'max-w-80')}
+          variant={ButtonVariant.Float}
+          onClick={onRequestClose}
+        >
+          Close
+        </Button>
       </Modal.Body>
     </Modal>
   );
```

Reply:
Yeah guess it's easier in case they change copy or something silly 🤪

Reply:
Changed to hide drawer button and use inline button instead.